contents/architecture/RESTArchitecturalStyle.md

```diff
@@ -0,0 +1,76 @@
+
+### <a name="uniform-interface"></a>Uniform Interface
+There should be a way for the server, the client and the intermediaries in the layers in between to communicate with each other. Hence, there should be a uniform interface in the system. The existence of the uniform interface is the foundation for the other 4 architectural constraints. Each component is encapsulated by the interface and hence become more independent of each other, allowing each to evolve independent of the rest. By having a uniform interface in the system, interactions between the layers can be monitored as the set of interactions are predefined. By allowing the interactions to be inspected by mediators (e.g., network firewalls), the security of the system is enhanced. However, the existence of the uniform interface might compromise the efficiency of the system as the information is transmitted in a standard format rather than catering to each component’s needs.
+
+There are four sub-constraints which further specify the Uniform Interface constraint.
+ * Identification of resources: as mentioned before (see [Meaning Of REST](#meaning-of-rest)), “resource” is an organization of information and the identifiers of the resources need to remain constant. An example to illustrate this constraint is [URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier). 
+ * Manipulation of resources through representations: as mentioned before (see [Meaning Of REST](#meaning-of-rest)), the “representation” of the resource at one point i.e. the representation of the state of the resource, is what the users see and modify, instead of the resource itself. 
+ * Self-descriptive messages: A message in a communication between the web components should contain all the information needed for the web components to understand its content. 
+ * Hypermedia as the engine of application state (HATEOAS): There should be hyperlinks embedded inside the representations given to the client, such that all the future actions that the client might take are within these representations. Hence, the client can interact with and navigate through the application without any prior knowledge of how to do so. Hence, the client and the server are more independent of each other. 
+
+### <a name="code-on-demand"></a>Code On Demand (Optional)
+The server can send a code snippet to the client to let the client execute. One example of this is the Javascript code sent along with the webpage in HTML. This constraint extends the client functionality and reduces the workload of the server by reducing the number of features to be implemented at the server. However, it reduces the visibility of the interactions between the client and the server and makes it more difficult to monitor the interactions. Hence, this constraint might be disabled in some implementations in REST style.
+
```

Comment:
You can consider merging some of the sections to form these three main sections:
>## What is REST
>## Why REST
>## How REST Works

Reply:
I am actually more inclined to my current structure. After merging the sections, the structure will look like this:
## What is REST
### Definition
### Meaning of REST
### Background of REST (why REST exists)
## Why learn REST
## How REST Works

I feel that before the reader sees the reason why he should continue reading this (dry) article i.e. Why learn REST, he might give up halfway on trying to understand REST :( That's why i feel that I should give a one-liner introduction then (try to) hook the reader with a reason why he should continue